analyzers/src/SonarAnalyzer.Common/Extensions/RegexExtensions.cs

```diff
@@ -24,6 +24,21 @@ namespace SonarAnalyzer.Extensions;
 
 public static class RegexExtensions
 {
+    /// <summary>
+    /// Matches the input to the regex. Returns <see cref="Match.Empty" /> in case of an <see cref="RegexMatchTimeoutException" />.
+    /// </summary>
+    public static Match SafeMatch(this Regex regex, string input)
```

Comment:
Looks weird, but it's because I moved a method down.
Look at the file entirely in both master and the branch, should be easy to understand what is happening.